.github/workflows/c-cpp.yml

```diff
@@ -21,22 +21,26 @@ jobs:
 
   build:
     runs-on: ubuntu-latest
-    container: niuslar/stm32-cubeide-headless:latest
+    container: 
+      image: ghcr.io/niuslar/stm32-cubeide-headless:latest
+      credentials:
```

Comment:
Why do we need credentials to run this?

Reply:
Hmm, it seems that we don't. I'll remove those. It even works without credentials when the image is private, I tried on another repo. 
I also gotta fix some other thing, I realized that over the weekend, it will take a day or two.

Reply:
Hi, I removed the credentials, ready to be merged when you are. (I will open another PR for the other pipeline related stuff that I wanted to polish)